mesos/provider.go

```diff
@@ -41,24 +46,30 @@ func (p *ServiceProvider) buildServices(t task) ([]consul.ServiceInstance, error
 	}
 
 	var services []consul.ServiceInstance
-	var tags []string
+	var globalTags []string
 
 	for _, label := range t.Labels {
-		if label.Value == "tag" {
-			tags = append(tags, label.Key)
+		if label.Value == consulTagValue {
+			globalTags = append(globalTags, label.Key)
 		}
 	}
 
+	marathonTaskTag := fmt.Sprintf("marathon-task:%s", t.ID)
+	globalTags = append(globalTags, marathonTaskTag)
+	tagPlaceholders := getPlaceholders(t.Discovery.Ports.Ports)
+
 	// TODO(medzin): add check conversion after MESOS-8780 is completed
 	// See: https://issues.apache.org/jira/browse/MESOS-8780
 	for _, port := range t.Discovery.Ports.Ports {
-		if consulServiceName := p.getConsulServiceName(port.Labels); consulServiceName != "" {
+		if consulServiceName := p.getConsulServiceName(port.Labels.Labels); consulServiceName != "" {
+			portTags := p.getPortLabels(port.Labels.Labels, tagPlaceholders)
+
 			service := consul.ServiceInstance{
 				ID:   fmt.Sprintf("%s_%d", hostname, port.Number),
 				Name: consulServiceName,
 				Host: hostname,
 				Port: port.Number,
-				Tags: tags,
+				Tags: append(portTags, globalTags...),
```

Comment:
mesos-executor and marathon-consul resolve placeholders also in globalTags. Here we don't resolve placeholders in global tags, which is inconsistent.